parsimonious/grammar.py

```diff
@@ -246,7 +247,7 @@ def _expressions_from_rules(self, rule_syntax, custom_rules):
     # A subsequent equal sign is the only thing that distinguishes a label
     # (which begins a new rule) from a reference (which is just a pointer to a
     # rule defined somewhere else):
-    label = ~"[a-zA-Z_][a-zA-Z_0-9]*" _
+    label = ~"[a-zA-Z_][a-zA-Z_0-9]*(?![\"'])" _
```

Comment:
This fixes the precedence bug for `u/r/b` bytestrings. This is a negative-lookahead assertion in the regex, preventing the rule from matching if it's immediately followed by a quote character.

I considered making this a minimal change to only disallow labels named u, r or b right before a string literal. However, it seemed to add more complexity with a pretty small benefit. Adding a space between references and literals looks nicer, and is how it works in all docs / examples in the library.